server/aws-lsp-codewhisperer/src/language-server/securityScan/securityScanDiagnosticsProvider.ts

```diff
@@ -23,10 +23,10 @@ class SecurityScanDiagnosticsProvider {
         this.findings = []
     }
 
-    resetDiagnostics() {
-        this.findings.forEach(finding => {
-            this.logging.log(finding.filePath)
-            this.publishDiagnostics(finding.filePath, [])
+    async resetDiagnostics() {
+        this.findings.forEach(async finding => {
+            this.logging.log(`reset diagnostics for: ${finding.filePath}`)
+            await this.publishDiagnostics(finding.filePath, [])
```

Comment:
What's the expected behavior of the `forEach` loop here? Is the function supposed to wait until all the publish diagnostics execution are complete?

If so, we may need to switch to `await Promise.all(this.findings.map(...))` . 
If not, the async/await keywords here doesn't change the behavior much; it will still fire multiple async call and move forward in the function

Reply:
I don't think we want to for all findings publish diagnostic to finish but the intent is we want to wait for the resetDiagnostic call to finish before we return from the didChangeWorkspaceFolder handler